src/Cpp/1-getting-started/14-HelloCOM/HelloCOM.cpp

```diff
@@ -0,0 +1,330 @@
+    const VertexFormat vertices[] =
+    {
+        //         Vertex                  Color
+        { {  0.0f,  0.5f, 0.0f }, { 0.25f, 0.39f, 0.19f } },
+        { {  0.5f, -0.5f, 0.0f }, { 0.44f, 0.75f, 0.35f } },
+        { { -0.5f, -0.5f, 0.0f }, { 0.38f, 0.55f, 0.20f } },
+    };
+    D3D11_BUFFER_DESC bufferInfo = {};
+    bufferInfo.ByteWidth = sizeof vertices;
+    bufferInfo.Usage = D3D11_USAGE_IMMUTABLE;
+    bufferInfo.BindFlags = D3D11_BIND_VERTEX_BUFFER;
+    D3D11_SUBRESOURCE_DATA resourceData = {};
+    resourceData.pSysMem = vertices;
+    if (FAILED(_device->CreateBuffer(
+        &bufferInfo,
+        &resourceData,
+        &_triangleVerts)))
+    {
+        std::cout << "D3D11: Failed to create triangle vertex buffer\n";
+        return false;
+    }
+    const uint32_t indices[] =
+    {
+        0, 1, 2
+    };
+    bufferInfo.ByteWidth = sizeof indices;
+    bufferInfo.BindFlags = D3D11_BIND_INDEX_BUFFER;
+    resourceData.pSysMem = indices;
+    if (FAILED(_device->CreateBuffer(
+        &bufferInfo,
+        &resourceData,
+        &_triangleIndices)))
+    {
+        std::cout << "D3D11: Failed to create index buffer\n";
+        return false;
+    }
+    return true;
+}
+
+void HelloCOMApplication::OnResize(GLFWwindow* window, int32_t width, int32_t height)
+{
+    // The potential errors that happen in here should propagate somewhere maybe?
+    auto* application = static_cast<HelloCOMApplication*>(glfwGetWindowUserPointer(window));
+    application->_width = width;
+    application->_height = height;
+    application->_deviceContext->Flush();
+    application->_renderTarget.Reset();
+    if (FAILED(application->_swapChain->ResizeBuffers(
+        0,
+        width,
+        height,
+        DXGI_FORMAT_B8G8R8A8_UNORM_SRGB,
+        0)))
+    {
+        std::cout << "D3D11: Failed to recreate SwapChain buffers\n";
+        return;
+    }
+    ComPtr<ID3D11Texture2D> backBuffer = nullptr;
+    if (FAILED(application->_swapChain->GetBuffer(
+        0,
+        __uuidof(ID3D11Texture2D),
+        &backBuffer)))
+    {
+        std::cout << "D3D11: Failed to acquire back buffer from the SwapChain\n";
+        return;
+    }
+    if (FAILED(application->_device->CreateRenderTargetView(
+        backBuffer.Get(),
+        nullptr,
+        &application->_renderTarget)))
+    {
+        std::cout << "D3D11: Failed to create Render Target from the back buffer\n";
+    }
+}
+
+void HelloCOMApplication::Update()
+{
+}
+
+void HelloCOMApplication::Render()
+{
+    D3D11_VIEWPORT viewport = {};
+    viewport.TopLeftX = 0;
+    viewport.TopLeftY = 0;
+    viewport.Width = _width;
+    viewport.Height = _height;
+    viewport.MinDepth = 0.0f;
+    viewport.MaxDepth = 1.0f;
+
+    const float clearColor[] = { 0.1f, 0.1f, 0.1f, 1.0f };
+    const UINT vertexStride = sizeof(VertexFormat);
+    const UINT vertexOffset = 0;
+
+    _deviceContext->OMSetRenderTargets(1, _renderTarget.GetAddressOf(), nullptr);
```

Comment:
Hmm can you check with 2-HelloFroge's deviceContext->XXX calls? i think i put them in the order of how its being used - pipeline wise, and I think it makes sense to show the user the same thing, might make things a little easier, once we explain dx11's render pipeline